src/west/app/project.py

```diff
@@ -924,6 +924,30 @@ def fetch_strategy(self, args):
         else:
             return 'smart'
 
+    def update_submodules(self, project):
+        # Updates given project submodules by using
+        # 'git submodule update --init --checkout --recursive' command
+        # from the project.path location.
+        submodules = project.submodules
+        call_location = os.path.join(self.topdir, project.path)
+
+        if submodules is None:
+            return
+
+        submodules_update_strategy = ('--rebase' if self.args.rebase
+                                      else '--checkout')
+        # For the list type, update given list of submodules.
+        if isinstance(submodules, list):
+            for submodule in submodules:
+                if isinstance(submodule, Submodule):
+                    project.git(['-C', call_location, 'submodule', 'update',
+                                 '--init', submodules_update_strategy,
+                                 '--recursive', submodule.path])
+        # For the bool type, update all project submodules
+        elif isinstance(submodules, bool):
```

Comment:
Does this run when submodules is False?

Reply:
Thanks, that was serious bug, I didn't spot it earlier, as it occurred after introducing some change suggested in our discussions. I changed check mentioned in this comment:

> Optimization nit: you could do this earlier. In fact you can probably do it before even calling this method which would then become a do_update_submodules() method (no need actually to change the name, just a sample name to get the point across).
>

from `if submodules is None:` to the `if not submodules` in order to consider also boolean False value. Moreover I also added new test, checking if submodules are not update when value is set to false, as such bug should not have happened.